libethereum/BlockChain.cpp

```diff
@@ -1160,20 +1160,38 @@ template <class K, class T> static unsigned getHashSize(unordered_map<K, T> cons
 void BlockChain::updateStats() const
 {
 	m_lastStats.memBlocks = 0;
-	DEV_READ_GUARDED(x_blocks)
-		for (auto const& i: m_blocks)
-			m_lastStats.memBlocks += i.second.size() + 64;
-	DEV_READ_GUARDED(x_details)
-		m_lastStats.memDetails = getHashSize(m_details);
-	DEV_READ_GUARDED(x_logBlooms)
+	{
```

Comment:
braces are not needed (`DEV_READ_GUARDED` makes the following statement a body of the loop with 1 iteration)

Reply:
So yeah, changes in `updateStats` don't seem to do anything

Reply:
Ah, OK.  It was not holding seven locks then.